test/unit/client-side-encryption/state_machine.test.ts

```diff
@@ -251,6 +251,41 @@ describe('StateMachine', function () {
       });
     });
 
+    context('when server closed the socket', function () {
+      let server;
+
+      beforeEach(async () => {
+        server = net.createServer(async socket => {
+          socket.end();
+        });
+        server.listen(0);
+        await once(server, 'listening');
+      });
+
+      afterEach(() => {
+        server.close();
+      });
+
+      it('throws a MongoCryptError error', async function () {
+        const stateMachine = new StateMachine({
+          proxyOptions: {
+            proxyHost: 'localhost',
+            proxyPort: server.address().port
+          }
+        } as any);
```

Comment:
Can we also test the case where just "endpoint" is set? then we would be covering the non-socks case.

Reply:
Added a host/port endpoint 👍